src/sources/aws_ecs_metrics/parser.rs

```diff
@@ -258,65 +258,102 @@ fn cpu_metrics(
-        vec![
-            ("usage_system_jiffies_total", cpu.system_cpu_usage),
-            (
-                "usage_usermode_jiffies_total",
-                cpu.cpu_usage.usage_in_usermode,
-            ),
-            (
-                "usage_kernelmode_jiffies_total",
-                cpu.cpu_usage.usage_in_kernelmode,
-            ),
-            ("usage_total_jiffies_total", cpu.cpu_usage.total_usage),
-            ("throttling_periods_total", cpu.throttling_data.periods),
-            (
-                "throttled_periods_total",
-                cpu.throttling_data.throttled_periods,
-            ),
-            (
-                "throttled_time_seconds_total",
-                cpu.throttling_data.throttled_time / 1_000_000_000.0,
-            ),
-        ]
-        .iter()
-        .map(|(name, value)| {
-            counter(
-                "cpu",
-                name,
-                namespace.clone(),
-                timestamp,
-                *value,
-                tags.clone(),
-            )
-        }),
-    );
+    if let Some(online_cpus) = cpu.online_cpus {
+        metrics.extend(vec![gauge(
+            "cpu",
+            "online_cpus",
+            namespace.clone(),
+            timestamp,
+            online_cpus as f64,
+            tags.clone(),
+        )]);
+    }
 
-    metrics.extend((0..cpu.online_cpus).filter_map(|index| {
-        cpu.cpu_usage.percpu_usage.get(index).map(|value| {
-            let mut tags = tags.clone();
-            tags.insert("cpu".into(), index.to_string());
+    if let Some(system_cpu_usage) = cpu.system_cpu_usage {
+        metrics.extend(vec![counter(
+            "cpu",
+            "usage_system_jiffies_total",
+            namespace.clone(),
+            timestamp,
+            system_cpu_usage,
+            tags.clone(),
+        )]);
+    }
 
-            counter(
-                "cpu",
-                "usage_percpu_jiffies_total",
-                namespace.clone(),
-                timestamp,
-                *value,
-                tags,
-            )
-        })
-    }));
+    if let Some(cpu_usage) = &cpu.cpu_usage {
+        metrics.extend(
+            vec![
+                ("usage_usermode_jiffies_total", cpu_usage.usage_in_usermode),
+                (
+                    "usage_kernelmode_jiffies_total",
+                    cpu_usage.usage_in_kernelmode,
+                ),
+                ("usage_total_jiffies_total", cpu_usage.total_usage),
+            ]
+            .iter()
+            .filter(|(_name, value)| value.is_some())
+            .map(|(name, value)| {
+                counter(
+                    "cpu",
+                    name,
+                    namespace.clone(),
+                    timestamp,
+                    value.unwrap(),
+                    tags.clone(),
+                )
+            }),
+        );
+    }
+
+    if let Some(throttling_data) = &cpu.throttling_data {
+        metrics.extend(
+            vec![
```

Comment:
I _think_ you could just use a slice here rather than a `Vec` since it is a static value:

```rust
&[
                  ("throttling_periods_total", throttling_data.periods),
// ...
]
```

Reply:
Hmmm the compiler did not like the `&` . It will let me just do `[ ...]` but I am looking into whether that is more efficient than `vec![...]`